DebugInfomationFormat.md

```diff
@@ -0,0 +1,220 @@
+#Debug Symbol Integration
+
+This is a design proposal for providing debug information within WebAssembly
+that would allow source-level debugging of code compiled into WebAssembly.
+
+## Goals and Principles
+
+The immediate goal is to implement enough debugging functionality in WebAssembly
+MVP to let users perform at least rudimentary debugging on their source.
+Another goal is not to bake in any limitations that would prevent reasonable
+future evolution of the standard.
+
+Specifically, we are currently not focused on identifying a language-independent
+subset of the debug format -- this proposal bundles language-dependent and
+language-independent parts together.
+
+META: We need an evolution strategy that allows new front-end/debugger pairs to
+use the format in the future to transfer information currently unanticipated.
+Examples: a) dynamic scoping in Lisp; b) full DWARF 4 equivalence.
+
+## Debug Info Goes into Extra Sections
+
+To allow easy stripping of debug info, all of it will go into a separate
+section.  In the binary format, this will be an unknown section inserted into
+the wasm file.  Because this new info will make the current
+[name section](BinaryEncoding.md#name-section) redundant, we propose deprecating
+the name section.
+
+META: How do we distinguish the debug-containing unknown section from any other
+unknown section?
+
+In the text format, this will be an extra child of the **module** element named
+**debug**.
+
+META: If there's a better name than **debug**, we can certainly use it.  Also,
+currently undecided about allowing multiple **debug** children, like **func**.
```

Comment:
Are you proposing that there be a single section, and a single format, that will suffice for all HLL debugging use cases? And perhaps further, that we attempt to build in the flexibility to grow to encompass everything DWARF can do, and everything needed to describe LISP, and anything else that comes up, all in this one format?